jekyll/_cci2/language-python.md

```diff
@@ -2,104 +2,266 @@
-3. To make changes you can edit the `.circleci/config.yml` file and make a commit. When you push a commit to GitHub, CircleCI will build and test the project.
+### Choose an Executor Type
 
-## Config Walkthrough
+A job's steps occur in a virtual environment called an [executor]({{ site.baseurl }}/2.0/executor-types/).
 
-Start with the version:
+In this example,
+the [`docker`]({{ site.baseurl }}/2.0/configuration-reference/#docker) executor is used
+to specify a custom Docker image.
+The first image listed becomes the job's [primary container]({{ site.baseurl }}/2.0/glossary/#primary-container).
+All of a job's commands execute in this container.
 
 ```yaml
-version: 2 # use CircleCI 2.0
+version: 2
+jobs:
+  build:
+    working_directory: ~/circleci-demo-python-django
+    docker:
+      - image: circleci/python:3.6.4  # primary container
 ```
 
-First, specify a `jobs` key. Each job represents a phase in your Build-Test-Deploy (BTD) process. The sample app only needs a `build` job, so all other configuration will be nested under that key.
+**Note:**
+`circleci/python:3.6.4` is a [convenience image]({{ site.baseurl }}/2.0/circleci-images/) provided by CircleCI.
+These images are extensions of official Docker images
+and include tools useful for CI/CD environments.
+
+### Add Other Services and Set Environment Variables
 
-Specify a working directory and container images for this build in `docker` section:
+You can specify additional containers for services like databases.
+Use the [`environment`]({{ site.baseurl }}/2.0/env-vars/#setting-an-environment-variable-in-a-container) key
+to set environment variables for all commands in a container.
 
 ```yaml
-...
-jobs: # a collection of steps
-  build: # runs not using Workflows must have a `build` job as entry point
-    working_directory: ~/circleci-demo-python-django # directory where steps will run
-    docker: # run the steps with Docker
-      - image: circleci/python:3.6.1 # ...with this image as the primary container; this is where all `steps` will run
-      - image: circleci/postgres:9.6.2 # database image for service container available at `localhost:<port>`
-        environment: # environment variables for database
+version: 2
+jobs:
+  build:
+    working_directory: ~/circleci-demo-python-django
+    docker:
+      - image: circleci/python:3.6.4
+        environment:
+          PIPENV_VENV_IN_PROJECT: true
+          DATABASE_URL: postgresql://root@localhost/circle_test?sslmode=disable
+      - image: circleci/postgres:9.6.2
+        environment:
           POSTGRES_USER: root
           POSTGRES_DB: circle_test
 ```
 
-Finally, add several `steps` within the `build` job:
+### Install Dependencies
+
+After choosing containers for a job,
+create [`steps`]({{ site.baseurl }}/2.0/configuration-reference/#steps) to run specific commands.
+
+Use the [`checkout`]({{ site.baseurl }}/2.0/configuration-reference/#checkout) step
+to check out source code.
+By default,
```

Comment:
Is this true? I think we have a new smart default for checkout that doesn't require working directory at all?

Reply:
Sort of. `working_directory` defaults to `~/project`. So you don't have to explicitly set `working_directory`.